apm-agent-plugins/apm-java-ldap-plugin/src/main/java/co/elastic/apm/agent/java_ldap/LdapClientAdvice.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.java_ldap;
+
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.GlobalTracer;
+import co.elastic.apm.agent.impl.transaction.AbstractSpan;
+import co.elastic.apm.agent.impl.transaction.Outcome;
+import co.elastic.apm.agent.impl.transaction.Span;
+import com.sun.jndi.ldap.Connection;
+import com.sun.jndi.ldap.LdapResult;
```

Comment:
Seems like IntelliJ has issues with the imports from the `java.naming/com.sun.jndi.ldap` module.


<img width="857" alt="Screen Shot 2022-01-28 at 13 52 13" src="https://user-images.githubusercontent.com/2163464/151550243-dbdd04df-7d11-4e96-aa2e-2871f663ebf8.png">

Applying the suggestion only works until the maven project is reloaded.

I've tried adding
```xml
    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <configuration>
                    <compilerArgs>
                        <arg>--add-exports</arg>
                        <arg>java.naming/com.sun.jndi.ldap=ALL-UNNAMED</arg>
                    </compilerArgs>
                </configuration>
            </plugin>
        </plugins>
    </build>
```

But that yields this error: `java: exporting a package from system module java.naming is not allowed with --release`

@tobiasstadler Did you manage to get this work in an IDE?

Reply:
Can you please try adding <maven.compiler.source>1.7</maven.compiler.source> to the properties of the plugin pom?

Reply:
The problem probably is that the `IntelliJ` profile in the parent pom sets `maven.compiler.target` and `maven.compiler.target` to 11.

Reply:
I just tested it with Java17. It compiles fine but the tests are failing with the error you mentioned.

Reply:
#2430